Sources/Conformance/conformance.pb.swift

```diff
@@ -245,18 +245,24 @@ struct Conformance_ConformanceRequest: SwiftProtobuf.Message, SwiftProtobuf.Prot
 
   init() {}
 
-  public mutating func _protoc_generated_decodeField<T: SwiftProtobuf.FieldDecoder>(setter: inout T, protoFieldNumber: Int) throws {
-    switch protoFieldNumber {
-    case 1, 2: try payload.decodeField(setter: &setter, protoFieldNumber: protoFieldNumber)
-    case 3: try setter.decodeSingularField(fieldType: Conformance_WireFormat.self, value: &requestedOutputFormat)
+  public mutating func _protoc_generated_decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
```

Comment:
Hmm. Fields that are arrays or maps can be stored in the message itself unconditionally, since they're already CoW types. For simple scalars, I would hope that there wouldn't be any additional retain/release overhead to reach into the class and pull the values out, but based on some of the generated assembly I saw when I was submitting a `Character` change PR to Swift stdlib, you might be right that they don't get optimized away very well.

We should profile it and see what we end up with, but it doesn't need to be done by 1.0. At the very least, it would be nice to figure out how to unify the code in the generator that writes these out so that there aren't two copies of it that need to be kept up-to-date.